File: 30-06-2025/p594.dart
class Solution {
  int findLHS(List<int> nums) {
    Map<int, int> freq = {};
    for (int n in nums) {
      freq[n] = (freq[n] ?? 0) + 1;
    }

    int ans = 0;
    for (int key in freq.keys) {
      if (freq.containsKey(key + 1)) {
        ans = max(ans, freq[key]! + freq[key + 1]!);
      }
    }
    return ans;
  }
}

File: 30-06-2025/p594.py
class Solution(object):
    def findLHS(self, nums):
        from collections import Counter
        freq = Counter(nums)
        max_len = 0
        for key in freq:
            if key + 1 in freq:
                max_len = max(max_len, freq[key] + freq[key + 1])
        return max_len